plugins/uefi/fu-plugin-uefi.c

```diff
@@ -613,40 +620,120 @@ load_custom_esp (FuPlugin *plugin, GError **error)
-					       data->esp_path);
+#endif
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_uefi_delete_old_capsules (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	g_autofree gchar *pattern = NULL;
+	g_autoptr(GPtrArray) files = NULL;
+
+	/* delete any files matching the glob in the ESP */
+	files = fu_common_get_files_recursive (data->esp_path, error);
+	if (files == NULL)
+		return FALSE;
+	pattern = g_build_filename (data->esp_path, "EFI/*/fw/fwupdate-*.cap", NULL);
+	for (guint i = 0; i < files->len; i++) {
+		const gchar *fn = g_ptr_array_index (files, i);
+		if (fnmatch (pattern, fn, 0) == 0) {
+			g_autoptr(GFile) file = g_file_new_for_path (fn);
+			g_debug ("deleting %s", fn);
+			if (!g_file_delete (file, NULL, error))
+				return FALSE;
+		}
 	}
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_uefi_delete_old_efivars (FuPlugin *plugin, GError **error)
+{
+	g_autoptr(GPtrArray) files = NULL;
 
+	/* delete any efivars matching the glob */
+	files = fu_common_get_files_recursive ("/sys/firmware/efi/efivars", error);
+	if (files == NULL)
+		return FALSE;
+	for (guint i = 0; i < files->len; i++) {
+		const gchar *fn = g_ptr_array_index (files, i);
+		if (fnmatch ("fwupdate-*", fn, 0) == 0) {
+			g_autoptr(GFile) file = g_file_new_for_path (fn);
+			g_debug ("deleting %s", fn);
+			if (!g_file_delete (file, NULL, error))
+				return FALSE;
+		}
+	}
 	return TRUE;
 }
 
 gboolean
-fu_plugin_coldplug (FuPlugin *plugin, GError **error)
+fu_plugin_startup (FuPlugin *plugin, GError **error)
 {
 	FuPluginData *data = fu_plugin_get_data (plugin);
-	fwup_resource *re;
-	gint supported;
-	g_autoptr(fwup_resource_iter) iter = NULL;
-	g_autofree gchar *name = NULL;
-	const gchar *ux_capsule_str = "Disabled";
 
-	/* supported = 0 : ESRT unspported
-	   supported = 1 : unlocked, ESRT supported
-	   supported = 2 : it is locked but can be unlocked to support ESRT
-	   supported = 3 : it is locked, has been marked to be unlocked on next boot
-			   calling unlock again is OK.
-	 */
-	supported = fwup_supported ();
-	if (supported == 0) {
+	/* get the supported status */
+	data->esrt_status = fwup_supported ();
+	if (data->esrt_status == FWUP_SUPPORTED_STATUS_UNSUPPORTED) {
 		g_set_error_literal (error,
 				     FWUPD_ERROR,
 				     FWUPD_ERROR_NOT_SUPPORTED,
 				     "UEFI firmware updating not supported");
 		return FALSE;
 	}
 
-	if (supported == 2) {
+	/* load any overriden options */
+	if (!fu_plugin_uefi_set_custom_mountpoint (plugin, error))
+		return FALSE;
+
+	/* get the default compiled-in value for the ESP mountpoint */
+#ifdef HAVE_FWUP_GET_ESP_MOUNTPOINT
+	if (data->esp_path == NULL)
+		data->esp_path = g_strdup (fwup_get_esp_mountpoint ());
+#endif
+
+	/* delete any existing .cap files to avoid the small ESP partition
+	 * from running out of space when we've done lots of firmware updates
+	 * -- also if the distro has changed the ESP may be different anyway */
+	if (data->esp_path != NULL) {
```

Comment:
To make this useful to < fwupdate 11 I wonder if it would be worth just checking and cleaning `/boot/efi` in the common scenario of user didn't set custom ESP mount point.

Reply:
Makes sense to me, fixed.